### PR TITLE
feat(Apple Podcasts): add Presence

### PR DESCRIPTION
**Pull request by 1RB**
## Description 
add presence for apple podcasts

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

image

</details>

### CODE REVIEW

File: websites/A/Apple Podcasts/presence.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+		},
+		strings = await presence.getStrings({
+			play: "general.playing",
+			pause: "general.paused",
+		}),
+		[privacy] = await Promise.all([presence.getSetting("privacy")]),
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:20]: useless promise.all if there is only 1 promise

File: websites/A/Apple Podcasts/presence.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		shadowRootDir = document.querySelector(".lcd").shadowRoot;
+	// document.querySelectorAll(".player > .artwork-container > .artwork-component > picture > img")[0].currentSrc
+	if (shadowRootDir?.querySelector(".lcd__active").ariaHidden !== "true") {
+		presenceData.type = ActivityType.Listening;
+
+		const episodeImage = (
+				shadowRootDir.querySelectorAll(
+					".lcd__artwork > picture > img"
+				)[0] as HTMLImageElement
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:29]:
```suggestion
				shadowRootDir.querySelector<HTMLImageElement>(
					".lcd__artwork > picture > img"
				)
```

File: websites/A/Apple Podcasts/presence.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+			startTimestamp: browsingTimestamp,
+		},
+		strings = await presence.getStrings({
+			play: "general.playing",
+			pause: "general.paused",
+		}),
+		[privacy] = await Promise.all([presence.getSetting("privacy")]),
+		shadowRootDir = document.querySelector(".lcd").shadowRoot;
+	// document.querySelectorAll(".player > .artwork-container > .artwork-component > picture > img")[0].currentSrc
+	if (shadowRootDir?.querySelector(".lcd__active").ariaHidden !== "true") {
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:23]: this comment needed here?

File: websites/A/Apple Podcasts/presence.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+		const episodeImage = (
+				shadowRootDir.querySelectorAll(
+					".lcd__artwork > picture > img"
+				)[0] as HTMLImageElement
+			).src.replace("88x88", "450x450"),
+			playingShadow = document.querySelector(".chrome-player").shadowRoot;
+
+		presenceData.name = shadowRootDir.querySelectorAll(
+			".lcd-meta-line__fragment"
+		)[0].textContent;
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:35]: Don't use querySelectorAll if you are just going to get the first item, then just use querySelector

File: websites/A/Apple Podcasts/presence.ts [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+		presenceData.largeImageKey = episodeImage;
+		const progress = shadowRootDir.querySelectorAll("#playback-progress")[0],
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:37]: ditto

File: websites/A/Apple Podcasts/presence.ts [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+			presenceData.smallImageKey = Assets.Pause;
+			presenceData.smallImageText = strings.pause;
+		}
+		presenceData.details = shadowRootDir.querySelectorAll(
+			".lcd-meta-line__fragment"
+		)[2].textContent;
+		presenceData.state = "on Apple Podcasts";
+	} else if (document.location.pathname.includes("/podcast")) {
+		presenceData.details = `Might listen to ${
+			document.querySelectorAll(".headings > .headings__title")[0].textContent
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:63]: ditto

File: websites/A/Apple Podcasts/presence.ts [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+		}`;
+		presenceData.state = `by ${
+			document.querySelectorAll(".headings > .headings__subtitles")[0]
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:66]: ditto

File: websites/A/Apple Podcasts/presence.ts [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+				.textContent
+		}`;
+		presenceData.largeImageKey = (
+			document.querySelector("picture > img") as HTMLImageElement
```
Comment (Bas950) [inline websites/A/Apple Podcasts/presence.ts:70]:
```suggestion
			document.querySelector<HTMLImageElement>("picture > img")
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by 1RB** [comment]
# updated images

| Setting | Result |
| --- | --- |
|image| image |
|image|image|
|image|image|


|image
| --- | 
|image|

File: package-lock.json
Comment (theusaf) [inline package-lock.json:1]: Undo changes to this file

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**